Sources/Sentry/Public/SentrySpanProtocol.h

```diff
@@ -87,6 +87,28 @@ NS_SWIFT_NAME(Span)
  */
 - (void)removeTagForKey:(NSString *)key NS_SWIFT_NAME(removeTag(key:));
 
+- (void)setMeasurement:(NSString *)name
```

Comment:
So apparently ObjC has the same issues as Dart, on Dart, we've used a single `enum` for now, and will refactor later once we are able to upgrade to the latest version.
Here we have an overload per unit type, not sure if it's good or bad, maybe just a method that takes a `String` would be enough.
Dart does not have overloads, so I could not do this either.
Maybe drop a message to folks that were part of the `setMeasurement` API and see if they are fine with that due to  the restrictions of the language?

The trade-off here is:
Exposing many public APIs, maybe more in the future, so it's overwhelming for the user.

Or a single `setMeasurement` method that takes a `unit` of the type `String` which is open and users would need to learn all the unit types thru the docs.
@sl0thentr0py and @jan-auer specifically were driver and approver.

Reply:
we said we wouldn't validate sdk side, so it's just a string in most sdks.

Reply:
Yep, but the implementation should be UX friendly, at least for strongly typed languages, an enum would be ideal, but one can set a `string` as well.
On java one can do:

```
transaction.setMeasurement("metric1", 2, MeasurementUnit.Duration.DAY)
transaction.setMeasurement("metric1", 2, MeasurementUnit.Custom("myOwnUnit"))
```

Reply:
yea that's fine, i think i added type hints to python as well

Reply:
sry forgot to follow up earlier, was on phone.
I have these type hints in python
https://github.com/getsentry/sentry-python/blob/f71a8f45e780525e52fa5868f45bb876dcf0994b/sentry_sdk/_types.py#L53-L82